### PR TITLE
examples: add an example of importing local lua code

### PR DESCRIPTION
**Pull request by cetanu**
This extends the existing example of using the Lua HTTP Filter in Envoy, so that there is a basic example of adding a lua file on the disk of the proxy, to be imported via a `require` statement in the inline code.

### CODE REVIEW

**Comment by zuercher** [comment]
Note this fails CI because `//test/config_test:example_configs_test` loads all the examples (to make certain they stay up-to-date) and can't find your module. I think you need to arrange for the module file to get added to the tar file in `test/configs/configgen.sh`.

**Comment by cetanu** [comment]
I've looked through these and don't see any obvious way to get this file added... Any hints?

**Comment by zuercher** [comment]
I haven't tried this, I think the following changes are sufficient:
1. `mylibrary.lua` needs to go into a subdirectory under `envoy/lua` (and update `lua/envoy.yaml` to match). Let's say that the library is now `examples/lua/lib/mylibrary.lua`. It doesn't matter much what the subdirectory's name is, but it will need to match what's done in step 3.
2. In `examples/BUILD`, add `lua/lib/mylibrary.lua` to the `configs` filegroup.
3. In `configs/configgen.sh`, we need to handle `*.lua` files the same way as `*.pem` files, so make "$OUT_DIR/lib" and add a `*.lua` entry to copy the files there. Again, the subdirectory needs to match step 1.

The tarball created by configgen.sh (normally found in bazel-bin/configs/example_configs.tar; you can run `bazel build //configs:example_configs` to generate it) should then contain your config file as `lua_envoy.yaml` and the library as `lib/mylibrary.lua`.

Hopefully that all makes sense. The subdirectory in step 1 is necessary because the example config test will try to load every file in the root directory of the tarball as an envoy config (regardless of file extension). Having the subdirectory names in steps 1 & 3 match makes it so the config can be run directly from the examples directory or from the test without modification.

File: examples/lua/envoy.yaml [hunk 1]
```diff
@@ -28,8 +28,10 @@ static_resources:
             typed_config:
               "@type": type.googleapis.com/envoy.config.filter.http.lua.v2.Lua
               inline_code: |
+                local mylibrary = require("mylibrary")
```
Comment (zuercher) [inline examples/lua/envoy.yaml]: I think this needs to `"lib/mylibrary"`.

File: examples/BUILD [hunk 2]
```diff
@@ -22,6 +22,7 @@ filegroup(
         "jaeger-tracing/service1-envoy-jaeger.yaml",
         "jaeger-tracing/service2-envoy-jaeger.yaml",
         "lua/envoy.yaml",
+	"lua/lib/mylibrary.lua",
```
Comment (zuercher) [inline examples/BUILD]: nit: spaces not tabs

File: examples/lua/Dockerfile-proxy [hunk 3]
```diff
@@ -1,2 +1,3 @@
 FROM envoyproxy/envoy-dev:latest
+ADD ./lib/mylibrary.lua mylibrary.lua
```
Comment (zuercher) [inline examples/lua/Dockerfile-proxy]: Also, I believe this needs to be
```
ADD ./lib/mylibrary.lua /lib/mylibrary.lua
```

**Comment by cetanu** [comment]
Sorry about that, I wasn't entirely sure how lua handles imports to nested paths... looks like the last commit may be okay. I hope.

**Comment by zuercher** [comment]
I think you’ll need to merge master or push an empty commit (`git commit --allow-empty’) to get the missing CI job to run again.

**Comment by cetanu** [comment]
🎉 thanks for helping to get the tests passing, hope this is useful for folks out there

**Pull request merged**